rosbag2_transport/src/rosbag2_transport/recorder.cpp

```diff
@@ -449,7 +449,11 @@ RecorderImpl::create_subscription(
     qos,
     [this, topic_name, topic_type](std::shared_ptr<const rclcpp::SerializedMessage> message) {
       if (!paused_.load()) {
-        writer_->write(message, topic_name, topic_type, node->get_clock()->now());
+        auto now = node->get_clock()->now();
+        // When using sim time, do not record messages before receiving first /clock
+        if (!(record_options_.use_sim_time && now.nanoseconds() == 0)) {
```

Comment:
```suggestion
        if (!(record_options_.use_sim_time && now.seconds() == 0 && now.nanoseconds() == 0)) {
```

Reply:
This is not necessary, it does not work the same as a `struct time_t`. `double seconds()` returns "seconds since epoch", and `int64_t nanoseconds()` returns "nanoseconds since epoch" - they are just different views into the same piece of information, and I prefer to use the integer value instead of incurring a division and trying to compare floats.

Reply:
Ok. this makes sense. Thanks for the explanation.